pkg/api/scylla/validation/cluster_validation.go

```diff
@@ -40,23 +40,94 @@ func ValidateScyllaCluster(c *scyllav1.ScyllaCluster) field.ErrorList {
 	return ValidateScyllaClusterSpec(&c.Spec, field.NewPath("spec"))
 }
 
+func ValidateUserManagedTLSCertificateOptions(opts *scyllav1.UserManagedTLSCertificateOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	return allErrs
+}
+
+func ValidateOperatorManagedTLSCertificateOptions(opts *scyllav1.OperatorManagedTLSCertificateOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	return allErrs
+}
+
+func ValidateTLSCertificate(cert *scyllav1.TLSCertificate, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	switch cert.Type {
+	case scyllav1.TLSCertificateTypeOperatorManaged:
+		if cert.OperatorManagedOptions != nil {
+			allErrs = append(allErrs, ValidateOperatorManagedTLSCertificateOptions(
+				cert.OperatorManagedOptions,
+				fldPath.Child("operatorManagedOptions"),
+			)...)
+		}
+
+	case scyllav1.TLSCertificateTypeUserManaged:
+		if cert.UserManagedOptions != nil {
+			allErrs = append(allErrs, ValidateUserManagedTLSCertificateOptions(
+				cert.UserManagedOptions,
+				fldPath.Child("userManagedOptions"),
+			)...)
+		} else {
+			allErrs = append(allErrs, field.Required(fldPath.Child("userManagedOptions"), ""))
+		}
+
+	case "":
+		allErrs = append(allErrs, field.Required(fldPath.Child("type"), ""))
+
+	default:
+		allErrs = append(
+			allErrs,
+			field.NotSupported(
+				fldPath.Child("type"),
+				cert.Type,
+				[]scyllav1.TLSCertificateType{
+					scyllav1.TLSCertificateTypeOperatorManaged,
+					scyllav1.TLSCertificateTypeUserManaged,
+				},
+			),
+		)
+
+	}
+
+	return allErrs
+}
+
+func ValidateAlternatorSpec(alternator *scyllav1.AlternatorSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if alternator.WriteIsolation != "" {
+		found := slices.ContainsItem(AlternatorSupportedWriteIsolation, alternator.WriteIsolation)
+		if !found {
+			allErrs = append(
+				allErrs,
+				field.NotSupported(
+					fldPath.Child("alternator", "writeIsolation"),
+					alternator.WriteIsolation,
+					AlternatorSupportedWriteIsolation,
+				),
+			)
+		}
+	}
+
+	if alternator.InsecureEnableHTTP != nil && *alternator.InsecureEnableHTTP {
+		if alternator.Port != 0 {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("port"), "deprecated port is not allowed to by specified"))
```

Comment:
```suggestion
			allErrs = append(allErrs, field.Forbidden(fldPath.Child("port"), "deprecated port is not allowed to be specified"))
```

Reply:
actually why is this **only** forbidden in conjunction with InsecureEnableHTTP? It feels a bit counterintuitive. Is this a step towards ignoring the port?

Reply:
the port is deprecated but we have to keep backwards compatibility, so we forbid it only in conjunction with the new options so it doesn't break storage migration (essentially when we know it's a manual edit using the new fields)